time timestamp
uint8  id
bool  state